website/src/app/blog/how-dns-based-traffic-routing-works/readme.mdx

```diff
@@ -0,0 +1,372 @@
+network of research institutions and government agencies. The system was
+designed with the assumption that all entities on the network were known and
+trusted, and that the network itself was secure.
+
+As the ARPANET grew to become the internet, however, this assumption no longer
+held.
+
+#### DNS spoofing
+
+<p className="text-sm italic text-center mx-auto">
+  Figure 4: An illustration of DNS spoofing
+</p>
+
+One of the more apparent security issues is that you have to trust the
+nameservers you're querying. If a malicious actor manages to compromise a
+nameserver, they can return false responses to your queries. Since your machine
+has no other way to verify the authoritative answer, it will happily connect to
+whatever IP address the malicious nameserver returned. This is known as a
+[DNS spoofing](https://en.wikipedia.org/wiki/DNS_spoofing) attack.
+
+In more recent years, various solutions have been created to address this and
+other shortcomings:
+[DNSSEC](https://en.wikipedia.org/wiki/Domain_Name_System_Security_Extensions)
+ensures the integrity of responses to your query, and DNS-over-TLS (DoT) and
+DNS-over-HTTPS (DoH) take this a step further and prevent eavesdroppers from
+seeing which queries you perform.
+
+#### DNS enumeration
+
+Ok, these solutions work great for protecting the answers to your queries, but
+what about protecting the nameservers from the queries themselves?
+
+If an organization defines records for its internal services, for example,
+anyone can query its authoritative nameservers to map out a list of all the
+organization's internal services. How convenient!
+
+This is known as **DNS enumeration**, and is typically one of the first steps
+taken by a malicious actor looking for potential entrypoints into an
+organization.
+
+Because of this issue, organizations often resort to running their own upstream
+resolvers, configured to return different results (or sometimes none at all)
+depending on the _source_ of the request. This technique is known as **Split
+DNS**.
+
+So how does it work?
+
+## How Split DNS works
+
+Split DNS is a technique wherein an orgnization maintains two (or more) separate
+DNS servers (or a single server configured with two "zones") -- one for internal
+resources and one for external resources. The internal server is only accessible
+to users who are connected to an organization's internal network (such as a VPN
+or branch office network), while the external zone is accessible to the outside
+world.
+
+<p className="text-sm italic text-center mx-auto">
+  Figure 5: Split DNS in action
+</p>
+
+As an example, let's say an organization has an internal service called
+`gitlab.company.com` which resolves to `10.10.10.10`. The organization's
+internal DNS server would be configured to respond to queries for this service,
+but the external DNS server would simply return `NXDOMAIN`, DNS speak for
+"Sorry, wrong number". This allows the organization to publish some records
```

Comment:
```suggestion
"Sorry, the number you have reached is not in service." This allows the organization to publish some records
```

"wrong number" makes me think someone picked up the phone, just not the person I expected. NXDOMAIN (AIUI) is more like, the call doesn't even reach any phone on the other end.